source_middleware.go

```diff
@@ -42,7 +42,9 @@ type SourceMiddlewareOption func(SourceMiddleware)
 func DefaultSourceMiddleware(opts ...SourceMiddlewareOption) []SourceMiddleware {
 	middleware := []SourceMiddleware{
 		&SourceWithSchemaExtraction{},
+		&SourceWithSchemaContext{},
```

Comment:
For a second I thought that we should turn these around, because we need `SourceWithSchemaContext` to be executed first, but then I noticed that `SourceWithMiddleware` actually wraps the middleware from first to last, so the last middleware in this list will be executed first.

What I'm trying to say is that this is fine now, but I'm thinking if we should rather change the for loop in line 59 to loop through the middleware in reverse order, so that they are executed in the order specified in the slice.

Maybe I'm overthinking this as well 😅

Reply:
What you're saying makes a lot of sense... However, I just swapped the places of the middleware and things still appear to be working (I see the schema context name in the destination :S). I'll look into how it works under the hood.